File: April_2022/11_04_2022/Correctness_of_printed_sheets.py
n = int(input())
data = list(map(int,input().split()))
if(data == sorted(data)):
    print("Correct")
else:
    flag = False
    m = 0
    for i in range(n):
        data.append(data.pop(0))
        if(data == sorted(data)):
            flag = True
            m = i+1

    if(flag):
        print("Easily correctable")
        print(m)
    else:
        print("Not easily correctable")

File: April_2022/11_04_2022/Correctness_of_printed_sheets.cpp
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;

int main(){

    int n;
    cin>>n;

    vector<int> data;
    vector<int> sorted;
    for(int i=0;i<n;i++){
        int temp;
        cin>>temp;
        data.push_back(temp);
        sorted.push_back(temp);
    }

    sort(sorted.begin(),sorted.end());
    if(data == sorted){
        cout<<"Correct"<<endl;
    }
    else{
        bool flag = false;
        int m = 0;
        for(int i=0;i<n;i++){
            int x = data[0];
            data.erase(data.begin());
            data.push_back(x);
            if(data == sorted){
                flag = true;
                m = i+1;
            }
        }
        if(flag){
            cout<<"Easily correctable\n"<<m<<endl;
        }
        else{
            cout<<"Not easily correctable"<<endl;
        }
    }

    return 0;
}